src/pymap/settings.py

```
@@ -213,85 +226,101 @@
+    data = None
+
+    if not config_file.is_file():
+        raise ImproperlyConfigured(f"Config path provided {config_file} is not a file")
+
+    with open(config_file) as f:
+        try:
+            data = json.load(f)
+        except (json.JSONDecodeError, UnicodeDecodeError) as e:
+            raise ImproperlyConfigured(
+                f"Failed to load json from config file: {config_file}, reason: {e}"
             )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)